TheEthicalHackerCup/Assets/DefenseMinigames/Ransomware/dRansomManager.cs

```diff
@@ -0,0 +1,190 @@
+using System.ComponentModel.Design;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Xml.Linq;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+public class dRansomManager : MonoBehaviour
+{
+    private string heuristics;
+    private IList<MetaData> metaData;
+    private int lives;
+    private int dataIndex;
+    private float timer;
+    private bool timerOn;
+    private MetaData current { get { return metaData[dataIndex]; } }
+
+    [SerializeField]
+    private TextMeshProUGUI fileText;
+
+    [SerializeField]
+    private TextMeshProUGUI livesText;
+
+    [SerializeField]
+    private TextMeshProUGUI timerText;
+
+    [SerializeField]
+    private TextMeshProUGUI sandboxStatusText;
+
+
+    [SerializeField]
+    private TextMeshProUGUI heuristicText;
+    [SerializeField]
+    private GameObject denyButton;
+    [SerializeField]
+    private GameObject acceptButton;
+    [SerializeField]
+    private GameObject sandboxButton;
+    
+
+    void Start()
+    {
+
+        //read and process files
+        var fileContent = Resources.Load<TextAsset>(FileHelperSingleton.Singleton.currentFileName).ToString();
+        Debug.Log(fileContent);
+        XDocument doc = XDocument.Parse(fileContent);
+        metaData = MetaData.ListFromXml(doc.Root.Element("MetaDataCollection"));
+        heuristics = doc.Root.Element("Heuristic").Value;
+
+
+
+        denyButton.GetComponent<Button>().onClick.AddListener(() => handleDenyClicked());
+        acceptButton.GetComponent<Button>().onClick.AddListener(() => handleAcceptClicked());
+        sandboxButton.GetComponent<Button>().onClick.AddListener(() => handleSandboxClicked());
+
+        lives = 4;
+        dataIndex = 0;
+        timer = 0;
+        timerOn = false;
+
+        updateHeuristics(heuristics);
+        updateAll();
+
+    }
+
+    private void updateHeuristics(string heuristics)
+    {
+        heuristicText.text = heuristics;
+    }
+
+    private void updateMetaData() 
+    {
+        fileText.text = current.ToString();
+    }
+
+    private void updateLives() {
+        this.livesText.text = "Lives: " + lives;
+    }
+
+    private void updateSandboxStatus() 
+    {
+        if(!this.timerOn) 
+        {
+            this.sandboxStatusText.text = "";
+            return;
+        }
+        sandboxStatusText.text = this.current.isVirus && this.current.TimeTillInfection <= timer?"Virus Detected":"Program Sandboxed";
+    }
+
+    private void updateTimer() {
+        this.timerText.text = timerOn ? "Time program is in sandbox: "+timer : "";
+    }
+
+    private void Update()
+    {
+        if (this.timerOn) 
+        {
+            timer += Time.deltaTime;
+            updateTimer();
+            updateSandboxStatus();
+        }
+    }
+
+    private void handleAcceptClicked() 
+    {
+        if (current.isVirus)
+        {
+            lives-=2;
+            updateLives();
+            checkForDeath();
+        }
+        nextData();
```

Comment:
Consider adding some text to the Heuristics in the XMLs that mentions that accepting viruses cost 2 lives

Reply:
itll be in the tutorial

Reply:
Since the tut would now be in a different scene, consider adding a pregame screen to your minigame because if a player goes to a tutorial and comes back to the scene just before completion, they will go though the Start() method again, restarting the game

Reply:
i dont think its that big of a deal

Reply:
but yeah sure its a small change